src/Std/Data/HashSet/Lemmas.lean

```diff
@@ -0,0 +1,274 @@
+/-
+Copyright (c) 2024 Lean FRO, LLC. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+prelude
+import Std.Data.HashMap.Lemmas
+import Std.Data.HashSet.Basic
+
+/-!
+# Hash set lemmas
+
+This file contains lemmas about `Std.Data.HashSet.Raw` and `Std.Data.HashSet`. Most of the lemmas require
+`EquivBEq α` and `LawfulHashable α` for the key type `α`. The easiest way to obtain these instances
+is to provide an instance of `LawfulBEq α`.
```

Comment:
Does it make sense to move all lemmas about `Raw` into a separate file? This makes it easier to direct users to the "outermost" law.

Reply:
Sorry, I don't understand the second sentence, could you elaborate?

Reply:
Sorry, typo on my part (layer -> law)

I just meant that since most users will not actually need to know about `Raw`, splitting these files in two would make the code more accessible.

Reply:
Yes, good idea. We also get a bit more build parallelism as an added bonus.